geonode/assets/handlers.py

```diff
@@ -38,10 +38,10 @@ def clone(self, asset: Asset) -> Asset:
     def create_link_url(self, asset: Asset) -> str:
         raise NotImplementedError()
 
-    def get_download_handler(self, asset: Asset) -> AssetDownloadHandlerInterface:
+    def get_download_handler(self, asset: Asset = None) -> AssetDownloadHandlerInterface:
```

Comment:
Why is it now optional? 
It's a generic framework, either the asset is always needed or it is not. 
I would leave it as it is.